presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -862,6 +863,10 @@ private boolean canRunMore()
                 return false;
             }
 
+            if (((RootInternalResourceGroup) root).isResourceLimitExceeded()) {
+                return false;
+            }
+
```

Comment:
I don't think this is the correct semantic. If `isResourceLimitExceeded` is expected to be a `root` level property, then we should just prevent running `internalStartNext` from `RootInternalResourceGroup.processQueuedQueries`. As of now I feel there is too much plumbing done through the resource groups which can be completely avoided.